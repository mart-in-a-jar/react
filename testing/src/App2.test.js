import { screen, render, act } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App2 from "./App2";

describe("App2 component", () => {
    test("Render magnificent monkeys", () => {
        const { container } = render(<App2 />);
        expect(container).toMatchSnapshot();
    });
    test("render radical rhinos after button click", async () => {
        const user = userEvent.setup();
        render(<App2 />);
        const button = screen.getByRole("button", { name: "Click Me" });
        await act(async () => {
            await user.click(button);
        });
        expect(screen.getByRole("heading").textContent).toMatch(
            /radical rhinos/i
        );
    });
});
