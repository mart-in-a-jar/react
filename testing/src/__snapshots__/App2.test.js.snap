// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App2 component Render magnificent monkeys 1`] = `
<div>
  <button
    type="button"
  >
    Click Me
  </button>
  <h1>
    Magnificent Monkeys
  </h1>
</div>
`;
